From: Evgeni Golov <evgeni@debian.org>
Date: Fri, 25 Dec 2015 12:02:08 +0100
Subject: sidebar-delimnullwide

SidebarDelim can be NULL and strlen(NULL) is a bad idea, as it will segfault.
Wrap it with NONULL().

While at it, change strlen to mbstowcs for better utf8 support.

Closes: #696145, #663883
---
 sidebar.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/sidebar.c b/sidebar.c
index e9d9eba..b75a0e3 100644
--- a/sidebar.c
+++ b/sidebar.c
@@ -30,6 +30,7 @@
 #include <libgen.h>
 #include "keymap.h"
 #include <stdbool.h>
+#include <wchar.h>
 
 /*BUFFY *CurBuffy = 0;*/
 static BUFFY *TopBuffy = 0;
@@ -150,7 +151,7 @@ char *make_sidebar_entry(char *box, unsigned int size, unsigned int new, unsigne
     struct sidebar_entry sbe;
     int SBvisual;
 
-    SBvisual = SidebarWidth - strlen(SidebarDelim);
+    SBvisual = SidebarWidth - mbstowcs(NULL, NONULL(SidebarDelim), 0);
     if (SBvisual < 1)
         return NULL;
 
@@ -193,7 +194,7 @@ int draw_sidebar(int menu) {
 #ifndef USE_SLANG_CURSES
         attr_t attrs;
 #endif
-        short delim_len = strlen(SidebarDelim);
+        short delim_len = mbstowcs(NULL, NONULL(SidebarDelim), 0);
         short color_pair;
 
         static bool initialized = false;
